### ISSUE TITLE
Invalid Login alert box while upgrading from 3.6.1 to 3.6.2

### ISSUE THREAD

**Issue by sajalsoni** [joomla/joomla-cms#11465]
#### Steps to reproduce the issue
Try to upgrade Joomla to 3.6.2 from 3.6.1 from the admin side.

#### Expected result
Should upgrade Joomla to 3.6.2

#### Actual result
Pops up alert box with "ERROR: Invalid Login"

#### System information (as much as possible)
Linux/Fedora

**Comment by gunjanpatel** [comment]
I have just tried to upgrade from 1.6.1 to 1.6.2 without any issues. Everything functions normally for me.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by DavidBoggitt** [comment]
Fine for me too.  I think you mean 3.6.1 to 3.6.2 as well @gunjanpatel :)

**Comment by RoterNagel** [comment]
PHP 5.3 is now to old. Thank you Joomla Developers :) how needs a warning for that?

You can login in /administrator and see the menu but nothing works anymore, 5.3 is the problem.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by gunjanpatel** [comment]
aah yes Thanks @DavidBoggitt. Corrected in my above comment.

**Comment by brianteeman** [comment]
@RoterNagel what makes you say 5.3 doesnt work - it should work fine

Yes it is VERY old and not supported and insecure but it should still be supported by Joomla<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by RoterNagel** [comment]
3.6.1 didn't work (/plugins/user/joomla/joomla.php => Line 217 and 225-236)
But I have to apologies with 3.6.2 it works again. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by PhilETaylor** [comment]
`Pops up alert box with "ERROR: Invalid Login"` 

This is ***not*** meaning a Joomla login, this is meaning the restoration script "password" is incorrect. This is a generated, stored, and consumed password you never see, and protects your site during the tiny window of opportunity when the upgrade (restoration from zip file) is taking place. 

This is part of the restore process, it means the restore ini file was not created or corrupt or other... just starting the process from scratch again should be enough to fix this issue

It can also alert like this if you click the back button AFTER a successful upgrade. Don't click the back button in the browser.

**Comment by step-ez** [comment]
I have tried to upgrade from 3.6.1 to 3.6.2 several times now and I get an error saying "The most recent request was denied because it contained an invalid security token. Please refresh the page and try again."  I have manually downloaded the upgrade patch and tried to manually install that as well.  At that time I noticed that it came back with my login credentials before starting the backup process.  However I use 2 factor authentication on my server and it never asked for the token to be put in.  Is there a way to get around this issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by PhilETaylor** [comment]
have you updated com_joomlaupdate? Search for it as an Extension Update and apply it before upgrading

Are you using Akeeba backup on update and have not updated akeeba backup today? There was a fix in Akeeba backup also released today

**Comment by step-ez** [comment]
Thank I noticed after a submitted the ticket that there were extension updates for Akeeba and Joomla updater.  I have updated them both and now the upgrade is working properly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/joomla-cms/11465</a>.</sub>

**Comment by PhilETaylor** [comment]
Excellent - please click close on this issue then :)

**Comment by brianteeman** [comment]
Closed

**Comment by sandstorm871** [comment]
I see this is closed, but I am experiencing this in a number of sites, that are updating from 3.6.2 to 3.6.4
Also sites that are already at 3.6.4, if I choose to reinstall Joomla core files I see this message too.
I cleared all cache, cleared update cache, checked everything else is updated first, etc?


Just wandering wether to create a new issue & refer/link back to this?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11465">issues.joomla.org/tracker/joomla-cms/11465</a>.</sub>

**Comment by zero-24** [comment]
Please show your screento be sure you have the same issue and not another.

**Comment by zero-24** [comment]
I have just saw this closed. Please open a new issue with the screenshot.

**Comment by PhilETaylor** [comment]
if the following file exists - delete it and then try again

`/administrator/components/com_joomlaupdate/restoration.php`

**Comment by sandstorm871** [comment]
That file exists.
I deleted it, tried again with same problem. Also the file is added again?
https://www.dropbox.com/s/75tblxi5800xed3/Screenshot%202016-11-07%2015.08.31.png?dl=0

**Comment by PhilETaylor** [comment]
what is the content of that file now - does it have any content ?

**Comment by sandstorm871** [comment]
Same filesize
I hadnt looked inside it before, but all seems to be relating to Akeeba Kickstart.

```
<?php
defined('_AKEEBA_RESTORATION') or die('Restricted access');
$restoration_setup = array(
	'kickstart.security.password' => 'REMOVED',
	'kickstart.tuning.max_exec_time' => '5',
	'kickstart.tuning.run_time_bias' => '75',
	'kickstart.tuning.min_exec_time' => '0',
	'kickstart.procengine' => 'direct',
	'kickstart.setup.sourcefile' => '/home/lowenerg/public_html/tmp/joomla_3-6-4-stable-update_package-zip?format=zip&jcompat=my.zip',
	'kickstart.setup.destdir' => '/home/lowenerg/public_html',
	'kickstart.setup.restoreperms' => '0',
	'kickstart.setup.filetype' => 'zip',
	'kickstart.setup.dryrun' => '0');
```

**Comment by PhilETaylor** [comment]
looks good to me. Have you updated the Joomla Update component to the very latest? (it has a separate release to the joomla release now) 

Install this https://github.com/joomla/joomla-cms/releases/download/3.6.1/com_joomlaupdate-3.6.1.zip and try again after deleting restoration.php

**Comment by mbabker** [comment]
The Akeeba references are expected, the update component uses some of the same code used in the Akeeba Kickstart engine.

**Comment by sandstorm871** [comment]
Thanks Phil I'll try this & report back, but its happening on a number of sites.
The one in the screenshot & with the copied restoration.php file is already running 3.6.4 , but other sites that I have at 3.6.2 are not updating, with this same issue.

**Comment by sandstorm871** [comment]
I updated with com_joomlaupdate 3.6.1 from the link you provided.
Deleted the restoration.php file & tried the Joomla Update > Reinstall Core Joomla Files again & get the same Invalid login popup message.

**Comment by zero-24** [comment]
Than please move this to an new issue else it got missed fast

**Comment by sandstorm871** [comment]
there is a PHP error log in administrator>components>com_joomlaupdate with this 
`[07-Nov-2016 09:34:20 CST6CDT] PHP Deprecated:  Methods with the same name as their class will not be constructors in a future version of PHP; Akeeba_Services_JSON_Error has a deprecated constructor in /home/lowenerg/staging/2/administrator/components/com_joomlaupdate/restore.php on line 925`
But I tested this already with PHP7.0.12 & PHP5.6

**Comment by mbabker** [comment]
Don't worry about that.  FWIW https://github.com/joomla/joomla-cms/pull/12385 fixes that but that alone isn't an issue yet.

**Comment by tmumm** [comment]
FWIW, I was seeing this same error message when trying to update some sites and turns out the issue was the hosting account was out of disk space - so the update file couldn't be downloaded and extracted.

**Comment by l-arnold** [comment]
Joomla Update at 3.6.2
Joomla at 3.7.2
Trying to go to 3.7.3

Invalid Login.
I had previously updated the admin password (a few weeks ago).  moved it back to original, but still same error.  "Invalid Login" in popup box.  Can't complete the upgrade.

**Comment by PhilETaylor** [comment]
`Invalid Login.`

This is NOTHING to do with the users/admins passwords at all. 

This pop up is about the internal secret login details that the Joomla update process creates before it starts an upgrade, and uses to authenticate requests during the upgrade process. 

restoration.php is normally located (and should be deleted if found) at /administrator/components/com_joomlaupdate/restoration.php

**Comment by l-arnold** [comment]
I've done that a few times, but error persists so wondering what is setting the PW itself.  It rebuilds the file each time and PW changes each time.

**Comment by PhilETaylor** [comment]
manually upgrade this time - this will fix the issue - and it will be fine for the future.

**Comment by l-arnold** [comment]
Thanks Phil!

**Comment by l-arnold** [comment]
Last week a 3.74 update went well.  Today however, I had to do another manual update  (3.74 to 3.75)...  strange thing.  Not quite sure why.  Another site is working fine.  Not sure why this is toggling in and out.
I realize this should not be a support site but thinking some correlation between the "admin" login and the "admin" login during an update would actually work.  Perhaps there is some XML password that is not matching current that perhaps I used a week ago.  Doesn't seem that I would have.

**Comment by Benevolent27** [comment]
I had this issue with upgrading Joomla 3.7.1 to 3.7.5 and I was able to somehow fix things and install it by doing the following:

1.  The first thing I noticed was a warning about my php.ini not having a temp directory set.. So I found the INI file and enabled the temporary directories in it  (I set it to "/tmp")  Then I restarted my Apache server.

The update was still failing with the same "invalid login" message.  Grr.

2.  So I tried manually downloading the update file and unpacking it over my Joomla install directory.  This seemed to do nothing.  I still got the same error.

3.  In an act of desperation, I tried setting 777 permission level to all directories in the Joomla installation folder.
`find /var/www/html/Joomla/MyJoomlaInstall -type d -exec sudo chmod 777 {} \;`
This made the "invalid login" error go away!  GREAT!  But then it got an error stating that it could replace a file and failed..

4.  So then I tried setting all files in the Joomla install directory to 777 permission..
'find /var/www/html/MyJoomlaInstall -type f -exec sudo chmod 777 {} \;'

And when I tried again.. Success!  Why this worked, I have literally no idea, but it did.  When done, you may want to rerun those permissions settings commands with 750 or whatever level you prefer.

![image](https://user-images.githubusercontent.com/8181775/29909019-0c34b8a2-8df2-11e7-9526-2f9b4633d813.png)

**Comment by bhaumikkotak** [comment]
I have upgrade the site from joomla 2.5.8 to 3.5.1

Now I am trying to upgrade from joomla 3.5.1 to 3.6.5 and I am facing the same issue 
ERROR:
Invalid login

And restoration.php file gets created again and again even if I delete it.

I have upgraded my com_joomlaupdate to the com_joomlaupdate-3.6.1 as well.

I also tried by using update package Joomla_3.6.5-Stable-Update_Package, but no luck, can any one please help me?

**Comment by franz-wohlkoenig** [comment]
@bhaumikkotak Please ask help on the [forums - Migrating and Upgrading to Joomla! 3.x](https://forum.joomla.org/viewforum.php?f=710). This repository concerns in first Place Joomla coding.
